File: app/src/main/java/com/example/ftpnext/core/LogManager.java
package com.example.ftpnext.core;

import android.util.Log;

public class LogManager {

    public static final String UNKNOWN_TAG = "UNKNOWN_TAG";

    private static final String TAG = "CORE - Log Manager";
    private static final String TAG_PREFIX = "FTPNEXT LOG : ";
    private static final boolean ENABLED_LOG = true;

    public static boolean error(String iMessage) {
        if (ENABLED_LOG)
            error(UNKNOWN_TAG, iMessage);
        return false;
    }

    public static boolean error(String iTAG, String iMessage) {
        if (ENABLED_LOG)
            Log.e(TAG_PREFIX, iTAG + " : " + iMessage);
        return false;
    }

    public static void info(String iMessage) {
        if (ENABLED_LOG)
            info(UNKNOWN_TAG, iMessage);
    }

    public static void info(String iTAG, String iMessage) {
        if (ENABLED_LOG)
            Log.i(TAG_PREFIX, iTAG + " : " + iMessage);
    }

}

File: app/src/main/java/com/vpulse/ftpnext/core/LogManager.kt
package com.vpulse.ftpnext.core

import android.util.Log

object LogManager {
    const val UNKNOWN_TAG = "UNKNOWN_TAG"
    private const val ENABLED_LOG = true
    private const val TAG = "CORE - Log Manager"
    private const val TAG_PREFIX = "FTPNEXT LOG : "
    private var sSilentDebugLog = false
    fun setSilentDebugLog(iValue: Boolean) {
        sSilentDebugLog = iValue
    }

    fun error(iMessage: String): Boolean {
        if (ENABLED_LOG) error(UNKNOWN_TAG, iMessage)
        return false
    }

    fun error(iTAG: String, iMessage: String): Boolean {
        if (ENABLED_LOG) Log.e(TAG_PREFIX, "$iTAG : $iMessage")
        return false
    }

    fun info(iMessage: String?) {
        if (ENABLED_LOG) info(UNKNOWN_TAG, iMessage)
    }

    fun info(iTAG: String, iMessage: String?) {
        if (ENABLED_LOG) Log.i(TAG_PREFIX, "$iTAG : $iMessage")
    }

    fun debug(iMessage: String) {
        if (ENABLED_LOG && !sSilentDebugLog) debug(UNKNOWN_TAG, iMessage)
    }

    fun debug(iTag: String, iMessage: String) {
        if (ENABLED_LOG && !sSilentDebugLog) Log.d(TAG_PREFIX, "$iTag : $iMessage")
    }

    fun verbose(iMessage: String) {
        if (ENABLED_LOG) verbose(UNKNOWN_TAG, iMessage)
    }

    fun verbose(iTag: String, iMessage: String) {
        if (ENABLED_LOG) Log.v(TAG_PREFIX, "$iTag : $iMessage")
    }
}